src/Util/MapUtil.js

```diff
@@ -255,6 +259,40 @@ export class MapUtil {
     return info;
   }
 
+  /**
+   * Get the getlegendGraphic url of a layer. Designed for geoserver.
+   * Currently supported Sources:
+   *  - ol.source.TileWms
+   *
+   * @param {ol.layer.Layer} layer The layer that you
+   */
+  static getLegendGraphicUrl(layer, extraParams) {
+    if (!(layer instanceof OlLayerBase)) {
+      Logger.error('Invalid input parameter for MapUtil.getLegendGraphicUrl.');
+      return;
+    }
+
+    if (layer instanceof OlLayerTile && layer.getSource() instanceof OlSourceTileWMS) {
+      const source = layer.getSource();
+      const url = source.getUrls()[0];
```

Comment:
What will happen if a layer has only `tileUrlFunction` defined? We should at least document the current expectation / limitation.